pkg/apis/core/types.go

```diff
@@ -2693,6 +2693,12 @@ type PodSpec struct {
 	// If not specified, the pod will not have a domainname at all.
 	// +optional
 	Subdomain string
+	// If set to true, the fully qualified domain name (FQDN) of a Pod is set as hostname of its containers.
+	// In Linux containers, this means setting the FQDN in the hostname field of the kernel (the nodename field of struct utsname).
+	// In Windows containers, this means setting the COMPUTERNAME to FQDN.
+	// If a pod does not have FQDN, this has no effect.
+	// +optional
+	HostnameFQDN bool
```

Comment:
Bringing conversation to this PR so we have all in one place

@tedyu:
Looking at existing bool fields, they start with a verb (such as Enable).

How about naming this field SetHostnameAsFQDN ?

@javidiaz:
I am open to change it. During the KEP review, I was suggested to use hostnameFQDN or fqdnHostname ( see kubernetes/enhancements#1792 (comment)). @thockin What do you think about changing the field as @tedyu suggest?

Reply:
I like SetHostnameAsFQDN, but I would like Tim to chime in here since he reviewed the KEP and I think he might do the API review.

Reply:
Fine

Reply:
thanks, changing to SetHostnameAsFQDN